src/layman/layer/geoserver/wfs.py

```diff
@@ -151,22 +151,24 @@ def clear_cache():
 
 def get_layer_info(workspace, layername, *, x_forwarded_items=None):
     uuid = layman_util.get_publication_uuid(workspace, LAYER_TYPE, layername)
-    return get_layer_info_by_uuid(uuid=uuid, x_forwarded_items=x_forwarded_items)
+    return get_layer_info_by_uuid(uuid=uuid, x_forwarded_items=x_forwarded_items, layman_workspace=workspace)
 
 
-def get_layer_info_by_uuid(*, uuid, x_forwarded_items=None):
+def get_layer_info_by_uuid(*, uuid, layman_workspace, x_forwarded_items=None):
     gs_layername = names.get_layer_names_by_source(uuid=uuid, ).wfs
-    wfs = get_wfs_proxy()
-    if wfs is None or uuid is None:
+    if uuid is None:
         return {}
     wfs_proxy_url = get_wfs_url(external_url=True, x_forwarded_items=x_forwarded_items)
 
-    wfs_layername = f"{gs_layername.workspace}:{gs_layername.name}"
-    if wfs_layername not in wfs.contents:
+    info = layman_util.get_publication_info_by_uuid(uuid, context={'keys': ['original_data_source']})
+    original_data_source = info['original_data_source']
+    data_store_name = get_external_db_store_name(uuid=uuid) if original_data_source == settings.EnumOriginalDataSource.TABLE.value else get_internal_db_store_name(db_schema=layman_workspace)
```

Comment:
I do not like, that the same 'get store name' logic is here in `get_layer_info_by_uuid` and in both method calling it (`refresh_wfs` and `patch_layer_by_uuid`).

Reply:
Encapsulated into get_db_store_name